src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala

```diff
@@ -245,87 +133,264 @@ class ScalaPageCreator(
                     }
                 }
                 case o => bdr.text(s"TODO: $o")
+        }
+
+        def buildBlock = (builder: ScalaPageContentBuilder#ScalaDocumentableContentBuilder) => builder
+            .group(kind = ContentKind.Cover, sourceSets = c.getSourceSets.asScala.toSet) { gbdr => gbdr
+                .cover(c.getName)()
+                .sourceSetDependentHint(Set(c.getDri), c.getSourceSets.asScala.toSet) { sbdr => 
+                    val s1 = sbdr
+                        .signature(c)
+                    co.fold(s1){ co => s1
+                        .group(kind = ContentKind.Symbol){ gbdr => gbdr
+                            .text("Companion ")
+                            .driLink(
+                                ext.kind match {
+                                    case dotty.dokka.Kind.Object => "class"
+                                    case _ => "object"
+                                },
+                                co
+                            )
+                        }
+                    }.contentForDescription(c)
+
+                }
             }
-            
+            .group(styles = Set(ContentStyle.TabbedContent)) { b => b
+                .contentForScope(c)
+                .groupingBlock(
+                    "Given",
+                    if(!givens.isEmpty) List(() -> givens.sortBy(_.getName).toList) else List.empty,
+                    kind = ContentKind.Functions
+                )( (bdr, splitter) => bdr ){ (bdr, elem) => bdr
+                    .driLink(elem.getName, elem.getDri, kind = ContentKind.Main)
+                    .sourceSetDependentHint(
+                        dri = Set(elem.getDri), 
+                        sourceSets = elem.getSourceSets.asScala.toSet, 
+                        kind = ContentKind.SourceSetDependentHint
+                    ){ srcsetbdr => srcsetbdr
+                        .contentForBrief(elem)
+                        .signature(elem)
+                    }
+                }
+                .groupingBlock(
```

Comment:
Can we split this rendering into few function calls using extension methods?

Reply:
Done